packages/design-system/src/components/text-field/types.ts

```diff
@@ -0,0 +1,40 @@
+import type {
+  TextFieldProps as AriaTextFieldProps,
+  TextFieldRenderProps as RACTextFieldRenderProps,
+} from 'react-aria-components';
+import type { PartialDeep } from 'type-fest';
+import type { InputClassNames } from '../../components';
+import type { AsType, RenderPropsChildren } from '../../types';
+
+export type TextFieldClassNames = PartialDeep<{
+  container: string;
+  description: string;
+  error: string;
+  input: InputClassNames;
+  label: string;
+  textField: string;
+}>;
+
+export type TextFieldSizes = 'sm' | 'lg';
+
+export type TextFieldMapping = {
+  description: Partial<Record<TextFieldSizes, string>>;
+  error: Partial<Record<TextFieldSizes, string>>;
+};
+
+type BaseTextFieldProps = {
+  children?: RenderPropsChildren<TextFieldRenderProps>;
```

Comment:
You don't need to redefine children unless you're changing the type. This type is the same as the children type from the RAC